docs-src/docs/usingwidgets.md

```diff
@@ -0,0 +1,163 @@
+
+In this example the function `say_hello` is called when a [PushButton](pushbutton.md) is pressed.
+
+~~~python
+from guizero import App, PushButton, Text
+
+def say_hello():
+    message.value = "Hello"
+
+app = App()
+message = Text(app)
+button = PushButton(app, text="Press me", command=say_hello)
+
+app.display()
+~~~
+
+![an app showing the text 'Hello` with a button underneath with the label 'Press me#](images/say-hello.png)
+
+In this example the `command` parameter of a [TextBox](textbox.md) widget is used to call the `count` function which displays how many characters have been entered.
+
+~~~python
+from guizero import App, TextBox, Text
+
+def count():
+    character_count.value = len(entered_text.value)
+
+app = App()
+entered_text = TextBox(app, command=count)
+character_count = Text(app)
+
+app.display()
+~~~
+
+![an app showing a TextBox with the word 'guizero' entered and the number 7 shown below](images/count-chars.png)
+
+### Controlling widgets
+
+The status of a widget can be controlled. This allows you to control how an app is used e.g. Only allowing the submit button of a form to be used when all the data has been entered
+
+You can control whether a widget is:
+
++ shown
++ hidden
++ enabled
++ disabled
+
+A widget will only appear on an app if it is visible. The widget can only be interacted with (used) if it is enabled.
+
+In this example [PushButton](pushbutton.md) widgets are used to change the status of a [TextBox](textbox.md) widget by calling the text box's `show`, `hide`, `disable` and `enable` methods.
+
+~~~python
+from guizero import App, PushButton, TextBox
+
+def show():
+    text_box.show()
+
+def hide():
+    text_box.hide()
+
+def enable():
+    text_box.enable()
+
+def disable():
+    text_box.disable()
+
+app = App()
+
+text_box = TextBox(app)
+
+show_button = PushButton(app, text="show", command=show)
+hide_button = PushButton(app, text="hide", command=hide)
+enable_button = PushButton(app, text="enable", command=enable)
+disable_button = PushButton(app, text="disable",command=disable)
+
+app.display()
+~~~
+
+### Destroying widgets
+
+Widgets can be destroyed (removed) from an application by using the `destroy` method. 
+
+Once a widget has been destroyed it cannot be used or referenced again. An error will be raised when trying to call a destroyed widget.
+
+~~~python
+from guizero import App, PushButton, Text
+
+def destroy_message():
+    message.destroy()
+
+app = App()
+message = Text(app, text="Destroy me!")
+destroy_button = PushButton(app, text="Destroy", command=destroy_message)
+
+app.display()
+~~~
+
+**Warning** - destroying a widget in an uncontrolled way could leave your application in an non-determinate state. A widget's execution is stopped immediately regardless of the state of your application. It is often to more appropriate to hide or disable a widget rather than destroying it.
```

Comment:
Maybe a bit of rewording to fit in with the more friendly style above: 

**Warning** - destroying a widget could leave your application in an non-determinate state. When you destroy a widget, its execution is stopped immediately regardless of what is going on in your application. It is usually safer to hide or disable a widget rather than destroying it completely.

Reply:
Yep.

Reply:
replaced with

```
**Warning** - destroying a widget in an uncontrolled way could make your application unstable. A widget's code will stop immediately regardless of what your application is doing. It is often to simpler and safer to hide or disable a widget rather than destroying it.
```